src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -279,6 +301,48 @@ private static void OnRequestSendingRequest(WinHttpRequestState state)
                 var serverCertificate = new X509Certificate2(certHandle);
                 Interop.Crypt32.CertFreeCertificateContext(certHandle);
 
+                IPAddress? ipAddress = null;
+                unsafe
+                {
+                    Interop.WinHttp.WINHTTP_CONNECTION_INFO connectionInfo;
+                    Interop.WinHttp.WINHTTP_CONNECTION_INFO* pConnectionInfo = &connectionInfo;
+                    uint infoSize = (uint)sizeof(Interop.WinHttp.WINHTTP_CONNECTION_INFO);
+                    if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(state, $"sizeof(WINHTTP_CONNECTION_INFO)={infoSize}");
+                    if (Interop.WinHttp.WinHttpQueryOption(
+                        state.RequestHandle,
+                        Interop.WinHttp.WINHTTP_OPTION_CONNECTION_INFO,
+                        (IntPtr)pConnectionInfo,
+                        ref infoSize))
+                    {
+                        ReadOnlySpan<byte> RemoteAddressSpan = new ReadOnlySpan<byte>(connectionInfo.RemoteAddress, 128);
+                        AddressFamily addressFamily = (AddressFamily)BitConverter.ToInt16(RemoteAddressSpan.ToArray(), 0);
+                        ipAddress = addressFamily switch
+                        {
+                            AddressFamily.InterNetwork => new IPAddress(BinaryPrimitives.ReadUInt32LittleEndian(RemoteAddressSpan.Slice(4))),
+                            AddressFamily.InterNetworkV6 => new IPAddress(RemoteAddressSpan.Slice(8, 16).ToArray()),
+                            _ => null
+                        };
+                        if (ipAddress != null && NetEventSource.Log.IsEnabled()) NetEventSource.Info(state, $"ipAddress: {ipAddress}");
```

Comment:
Do we really expecting non-IP address families here? Or can we just do `Debug.Assert()` for it and avoid further checks?

Reply:
I think we can just Debug.Assert this